### PR TITLE
Add config for chiado gnosis testnet

### PR DESCRIPTION
**Pull request by jmederosalvarado**
## Changes:
- Adds config for what will be gnosis first public testnet with merge support, named `chiado`

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Other (please describe): Config

## Testing
**Requires testing**

- [ ] Yes
- [x] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by MarekM25 — CHANGES_REQUESTED** [review]
Could you add chainspec? What's about denver network?

File: src/Nethermind/Nethermind.Runner/configs/chiado.cfg [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "Init": {
+    "MemoryHint": 768000000
+    "ChainSpecPath": "chainspec/chiado.json",
+  },
+  "Sync": {
+    "FastSync": true,
+    "FastBlocks": true,
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner/configs/chiado.cfg:8]: FastBlocks without pivot?

File: src/Nethermind/Nethermind.Runner/configs/chiado.cfg [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      "http://localhost:8551|http;ws|net;eth;subscribe;engine;web3;client"
+    ]
+  },
+  "AuRaMerge": {
+    "Enabled": true
+  },
+  "Aura": {
+    "TxPriorityContractAddress": "0x4100000000000000000000000000000000000000",
+    "ForceSealing": true,
+    "AllowAuRaPrivateChains": true
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner/configs/chiado.cfg:27]: is it a private chain now? I think not.

**Review by LukaszRozmej — COMMENTED** [review]
FastBlocks and AllowAuRaPrivateChains are probably to be removed?

**Comment by LukaszRozmej** [comment]
Should they be added to some tests?

**Comment by MarekM25** [comment]
@jmederosalvarado fix the runner tests ;)

File: src/Nethermind/Nethermind.Runner/configs/chiado.cfg (hunk 2 above)
Comment (dapplion) [reply src/Nethermind/Nethermind.Runner/configs/chiado.cfg:27]: Please adapt the flags to what makes sense, ansible config can be sloppy with those, lots of bagage from previous runs so don't copy what's there blindly

File: src/Nethermind/Nethermind.Runner/configs/denver.cfg [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{
+  "Init": {
+    "GenesisHash": "0x203e164cf3b6f6765abf2f0355a4d09e5eaf6a777243bc4407b569431cd95cb3",
+    "ChainSpecPath": "chainspec/denver.json",
+    "MemoryHint": 768000000
+  },
+  "Sync": {
+    "FastSync": true,
+    "FastBlocks": true,
+    "UseGethLimitsInFastBlocks": false,
```
Comment (dapplion) [inline src/Nethermind/Nethermind.Runner/configs/denver.cfg:10]: I don't really know what any of this options mean, please review to something that makes sense to you

File: src/Nethermind/Nethermind.Runner/configs/chiado.cfg [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  },
+  "Merge": {
+    "Enabled": true,
+    "TerminalTotalDifficulty": 115792089237316195423570985008687907853269984665640564039457584007913129639935
```
Comment (dapplion) [inline src/Nethermind/Nethermind.Runner/configs/chiado.cfg:31]: Warning! TTD for chiado is already scheduled `70560951604725799783765358597051456327101440` from https://github.com/gnosischain/consensus-deployment-ansible/tree/master/chiado#chiado-testnet

Comment (dapplion) [reply src/Nethermind/Nethermind.Runner/configs/chiado.cfg:31]: This TTD value is set twice, here and in the genesis file. Should you set it only in one place? Now it's inconsistent

File: src/Nethermind/Nethermind.Runner/configs/denver.cfg [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    "Enabled": true
+  },
+  "Aura": {
+    "TxPriorityContractAddress": "0x4100000000000000000000000000000000000000",
+    "ForceSealing": true,
+    "AllowAuRaPrivateChains": true
+  },
+  "Merge": {
+    "Enabled": true,
+    "TerminalTotalDifficulty": 115792089237316195423570985008687907853269984665640564039457584007913129639935
```
Comment (dapplion) [inline src/Nethermind/Nethermind.Runner/configs/denver.cfg:32]: Would be better to set only once, either here or in the genesis file

File: src/Nethermind/Nethermind.Runner/configs/denver.cfg [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+      "http://localhost:8551|http;ws|net;eth;subscribe;engine;web3;client"
+    ]
+  },
+  "AuRaMerge": {
+    "Enabled": true
+  },
+  "Aura": {
+    "TxPriorityContractAddress": "0x4100000000000000000000000000000000000000",
+    "ForceSealing": true,
+    "AllowAuRaPrivateChains": true
```
Comment (dapplion) [inline src/Nethermind/Nethermind.Runner/configs/denver.cfg:28]: Not a private chain

File: src/Nethermind/Nethermind.Runner/configs/denver.cfg [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    "Port": 8545,
+    "AdditionalRpcUrls": [
+      "http://localhost:8551|http;ws|net;eth;subscribe;engine;web3;client"
+    ]
+  },
+  "AuRaMerge": {
+    "Enabled": true
+  },
+  "Aura": {
+    "TxPriorityContractAddress": "0x4100000000000000000000000000000000000000",
```
Comment (dapplion) [inline src/Nethermind/Nethermind.Runner/configs/denver.cfg:26]: Why has this contract to be specified here?

Comment (jmederosalvarado) [reply src/Nethermind/Nethermind.Runner/configs/denver.cfg:26]: I copied it from [here](https://github.com/gnosischain/consensus-deployment-ansible/blob/043171e6cdf3e55e55c97b8f3933e3d1d7d6bc57/denver/custom_config_data/configs/nethermind_config.cfg#L16). is it not being used or smth?

Comment (dapplion) [reply src/Nethermind/Nethermind.Runner/configs/denver.cfg:26]: You tell me, all I know is that AuRa has many different contracts but only this one is specified here. I don't know the internals of Nethermind, nor which contracts have to be specified here and for what reason

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Runner/configs/denver.cfg:24]: So priorities contract is used for potentially sorting transactions when creating blocks. Priorities can come from contract or local configuration. Priorities are optional. Not sure if you deployed this contract on denver. More information can be found here: https://github.com/NethermindEth/nethermind/issues/2300

Comment (jmederosalvarado) [reply src/Nethermind/Nethermind.Runner/configs/denver.cfg:24]: Also. I see [here](https://github.com/gnosischain/consensus-deployment-ansible/blob/6357cf13b841c4420d6f691d19ce6629cd944a82/denver/custom_config_data/nethermind_genesis.json#L288-L291) that you have a contract deployed to this address in denver genesis. That's what is being used as `TxPriorityContract` in denver.

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Runner/configs/denver.cfg (hunk 7 above)
Comment (dapplion) [reply src/Nethermind/Nethermind.Runner/configs/denver.cfg:24]: Got it, thanks!

**Pull request merged**